src/alignment/distance.rs

```diff
@@ -57,32 +58,39 @@ pub fn hamming(alpha: TextSlice<'_>, beta: TextSlice<'_>) -> u64 {
 /// let ldist = levenshtein(x, y); // Distance is 5
 /// assert_eq!(ldist, 5);
 /// ```
-#[allow(unused_assignments)]
+#[allow(non_snake_case)]
 pub fn levenshtein(alpha: TextSlice<'_>, beta: TextSlice<'_>) -> u32 {
-    let mut columns = [vec![0u32; alpha.len() + 1], vec![0u32; alpha.len() + 1]];
-    let mut i_prev = 0;
-    let mut i_cur = 1;
+    let (m, n) = (alpha.len(), beta.len());
+    let mut M: Vec<u32> = Vec::with_capacity(m + 1); // the dynamic programming matrix (only 1 column stored)
+    let mut s: u32; // M[i - 1][j - 1]
+    let mut c: u32; // M[i - 1][j]
+    let mut a: u8; // alpha[i]
+    let mut b: u8; // beta[j]
```

Comment:
Again, personal preference, but just give the variables long names. There's no harm in renaming `M` to `matrix` (or `dp_matrix` or something else that is more descriptive than `M`) etc. `a` and `b` are probably fine though.

Reply:
Roger that